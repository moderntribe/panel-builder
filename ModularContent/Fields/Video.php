<?php


namespace ModularContent\Fields;
use ModularContent\OEmbedder;
use ModularContent\Panel;

/**
 * Class Video
 *
 * @package ModularContent\Fields
 *
 * A field for adding video embed URLs.
 */
class Video extends Text {
	/**
	 * @param string $url
	 * @param array $args Any additional args you want to pass to the oembed provider
	 *   Common args include:
	 *     - maxwidth
	 *     - maxheight
	 *
	 * Note that the oembed provider may not respect your arguments.
	 *
	 * @return object
	 */
	public static function get_data( $url, $args = array() ) {
		$oembed = new OEmbedder( $url, $args );
		return $oembed->get_oembed_data();
	}
}