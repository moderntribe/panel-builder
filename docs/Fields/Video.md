## Video

Functionally the same as a `Text` field, but the data returned is an `oembed`.

See `Text` for more details.