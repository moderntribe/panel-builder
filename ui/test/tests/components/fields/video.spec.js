/* global expect b:true */

import React from 'react';
import { shallow, mount } from 'enzyme';
import Video from 'components/fields/video';
import renderer from 'react-test-renderer';

describe('Video field', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<Video description="This is a test description" label="Some test label" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders as a <div>', () => {
		const wrapper = shallow(<Video />);
		expect(wrapper.type()).toEqual('div');
	});

	it('allows us to set props', () => {
		const wrapper = mount(<Video description="This is a test" label="Cat Video" name="cat_video" />);
		expect(wrapper.props().description).toEqual('This is a test');
		wrapper.setProps({ description: 'Change props for test' });
		expect(wrapper.props().description).toEqual('Change props for test');

		expect(wrapper.props().label).toEqual('Cat Video');
		wrapper.setProps({ label: 'New Label' });
		expect(wrapper.props().label).toEqual('New Label');

		expect(wrapper.props().name).toEqual('cat_video');
		wrapper.setProps({ name: 'new_video' });
		expect(wrapper.props().name).toEqual('new_video');
	});

	it('has a label', () => {
		const wrapper = shallow(<Video />);
		expect(wrapper.find('label').length).toEqual(1);
	});

	it('renders a text input', () => {
		const wrapper = shallow(<Video />);
		expect(wrapper.find('input[type="text"]').length).toEqual(1);
	});
});
