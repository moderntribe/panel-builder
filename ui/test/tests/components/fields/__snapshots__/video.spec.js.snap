exports[`Video field renders correctly 1`] = `
<div
  className="field panel-field">
  <label
    className="label panel-field-label">
    Some test label
    <span
      className="icon"
      onBlur={undefined}
      onClick={undefined}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined} />
  </label>
  <span
    className="panel-input-field inputContainer">
    <input
      className="video-url"
      name="modular-content-"
      onChange={[Function]}
      size="40"
      type="text"
      value="" />
  </span>
</div>
`;
